arangod/Replication/DatabaseInitialSyncer.cpp

```diff
@@ -488,19 +493,10 @@ Result DatabaseInitialSyncer::runWithInventory(bool incremental, VPackSlice dbIn
 
```

Comment:
Reaching this code would not trigger the guard.
In the case we are Child we will finish the batch.
We will also finish the batch in case of fail()
Both have not been possible in original Code.

Is it possible that we need to `cancel()` the guard under some circumstances here?

Reply:
Should the batch not be finished in any case, exactly as the code currently does?